Modules/Filtering/Smoothing/include/itkFFTDiscreteGaussianImageFilter.h

```diff
@@ -0,0 +1,189 @@
+{
+public:
+  ITK_DISALLOW_COPY_AND_MOVE(FFTDiscreteGaussianImageFilter);
+
+  /** Standard class type aliases. */
+  using Self = FFTDiscreteGaussianImageFilter;
+  using Superclass = DiscreteGaussianImageFilter<TInputImage, TOutputImage>;
+  using Pointer = SmartPointer<Self>;
+  using ConstPointer = SmartPointer<const Self>;
+
+  /** Method for creation through the object factory. */
+  itkNewMacro(Self);
+
+  /** Run-time type information (and related methods). */
+  itkTypeMacro(FFTDiscreteGaussianImageFilter, DiscreteGaussianImageFilter);
+
+  /** Image type information. */
+  using InputImageType = typename Superclass::InputImageType;
+  using OutputImageType = typename Superclass::OutputImageType;
+
+  /** Extract some information from the image types.  Dimensionality
+   * of the two images is assumed to be the same. */
+  using OutputPixelType = typename Superclass::OutputPixelType;
+  using OutputInternalPixelType = typename Superclass::OutputInternalPixelType;
+  using InputPixelType = typename Superclass::InputPixelType;
+  using InputInternalPixelType = typename Superclass::InputInternalPixelType;
+
+  /** Pixel value type for Vector pixel types **/
+  using InputPixelValueType = typename Superclass::InputPixelValueType;
+  using OutputPixelValueType = typename Superclass::OutputPixelValueType;
+
+  /** Extract some information from the image types.  Dimensionality
+   * of the two images is assumed to be the same. */
+  static constexpr unsigned int ImageDimension = Superclass::ImageDimension;
+
+  /** Type of the pixel to use for intermediate results */
+  using typename Superclass::RealOutputPixelType;
+  using typename Superclass::RealOutputImageType;
+  using typename Superclass::RealOutputPixelValueType;
+  using RealPixelType = RealOutputPixelType;
+  using RealImageType = RealOutputImageType;
+  using RealImagePointerType = typename RealImageType::Pointer;
+
+  /** Typedef to describe the boundary condition. */
+  using typename Superclass::BoundaryConditionType;
+  using typename Superclass::InputBoundaryConditionPointerType;
+  using typename Superclass::InputDefaultBoundaryConditionType;
+  using typename Superclass::RealBoundaryConditionPointerType;
+  using typename Superclass::RealDefaultBoundaryConditionType;
+
+  /** Typedef of double containers */
+  using typename Superclass::ArrayType;
+  using typename Superclass::SigmaArrayType;
+  using typename Superclass::ScalarRealType;
+
+  /** Typedef to describe kernel parameters */
+  using typename Superclass::KernelType;
+  using typename Superclass::RadiusType;
+
+  /** Typedef for convolution */
+  using ConvolutionImageFilterType = FFTConvolutionImageFilter<RealImageType, RealImageType, OutputImageType>;
+
+  /** Overridden accessors for unused parameters */
+
+  void
+  SetInputBoundaryCondition(const InputBoundaryConditionPointerType) override;
+
+  auto
+  GenerateKernelImage() -> RealImagePointerType;
```

Comment:
API should use raw pointers instead of smart pointers, primarily for wrapping simplification, ownership rules.

Reply:
Changes:
- Added `m_KernelImage` member so that the filter object has ownership of the kernel image, rather than previous method of passing ownership out of `GenerateKernelImage`;
- `GenerateKernelImage` is now a protected method and passes out a raw pointer reference to `m_KernelImage` after it has been initialized